Index: /home/thomas/projects/gostai/urbiengine-linux/trunk/.automerge
===================================================================
--- /home/thomas/projects/gostai/urbiengine-linux/trunk/.automerge	(revision 453)
+++ /home/thomas/projects/gostai/urbiengine-linux/trunk/.automerge	(working copy)
@@ -1,3 +1,2 @@
 393 | branches/1.0
-393 | branches/1.0/
 393 | branches/1.5
Index: /home/thomas/projects/gostai/urbiengine-linux/trunk/src/pc-server.hh
===================================================================
--- /home/thomas/projects/gostai/urbiengine-linux/trunk/src/pc-server.hh	(revision 453)
+++ /home/thomas/projects/gostai/urbiengine-linux/trunk/src/pc-server.hh	(working copy)
@@ -14,7 +14,7 @@
 # define PC_SERVER_HH_
 
 # include <libport/fwd.hh>
-# include <network/bsdnet/network.hh>
+# include <network/boost/network.hh>
 # include <kernel/userver.hh>
 
 namespace urbi
Index: /home/thomas/projects/gostai/urbiengine-linux/trunk/src/simul-api.cc
===================================================================
--- /home/thomas/projects/gostai/urbiengine-linux/trunk/src/simul-api.cc	(revision 453)
+++ /home/thomas/projects/gostai/urbiengine-linux/trunk/src/simul-api.cc	(working copy)
@@ -25,7 +25,7 @@
 #include <libport/unistd.h>
 
 // Inclusion order matters for windows. Leave userver.hh after network.hh.
-#include <network/bsdnet/network.hh>
+#include <network/boost/network.hh>
 #include <kernel/userver.hh>
 #include <kernel/uconnection.hh>
 
Index: /home/thomas/projects/gostai/urbiengine-linux/trunk/src/Makefile.am
===================================================================
--- /home/thomas/projects/gostai/urbiengine-linux/trunk/src/Makefile.am	(revision 453)
+++ /home/thomas/projects/gostai/urbiengine-linux/trunk/src/Makefile.am	(working copy)
@@ -33,7 +33,7 @@
 simul-api.hh simul-api.cc				\
 urbi-server.cc
 
-include $(top_srcdir)/src/network/bsdnet/bsdnet.mk
+include $(top_srcdir)/src/network/boost/boost.mk
 
 nodist_liburbicore_la_SOURCES = version.hh
 liburbicore_la_LIBADD =
Index: /home/thomas/projects/gostai/urbiengine-linux/trunk/src/urbi-server.cc
===================================================================
--- /home/thomas/projects/gostai/urbiengine-linux/trunk/src/urbi-server.cc	(revision 453)
+++ /home/thomas/projects/gostai/urbiengine-linux/trunk/src/urbi-server.cc	(working copy)
@@ -35,7 +35,7 @@
 const char* program_name = 0;
 
 // Inclusion order matters for windows. Leave userver.hh after network.hh.
-#include <network/bsdnet/network.hh>
+#include <network/boost/network.hh>
 
 #include <kernel/userver.hh>
 #include <kernel/uconnection.hh>
@@ -82,6 +82,7 @@
 	   "  -r        report the time taken by URBI loop to execute\n"
 	   "  -i        treat stdin as an urbi connection\n"
 	   "  -s PERIOD shell-mode (no network) with given period\n"
+	   "  -S, --ssl secured mode (URBI over SSL connection)\n"
 	   "  -t        use high priority scheduling\n"
 #ifndef NO_FAST_MODE
 	   "  -f        fast mode: the server will run as fast\n"
@@ -158,6 +159,8 @@
 	sa.freq = boost::lexical_cast<int>(argv[++i]);
 	sa.port = 0;
       }
+      else if (arg == "-S" || arg == "--ssl")
+	sa.secured = true;
       else if (arg == "-t")
 	sa.real_time = true;
       else if (arg == "-v" || arg == "--version")
@@ -215,6 +218,7 @@
     sa.address		= "localhost";
     sa.freq		= 0;
     sa.port		= UServer::TCP_PORT;
+    sa.secured          = false;
     sa.fast_mode	= false;
     sa.report_time	= false;
     sa.stdin_to_ghost	= false;
@@ -258,7 +262,7 @@
 
     if (sa.port!=-1)
     {
-      int port = Network::createTCPServer(sa.port, sa.address.c_str());
+      int port = Network::createTCPServer(sa.port, sa.address.c_str(), sa.secured);
       if (!port)
       {
 	std::cerr << "cannot bind to port " << sa.port
Index: /home/thomas/projects/gostai/urbiengine-linux/trunk/src/server-caller.hxx
===================================================================
--- /home/thomas/projects/gostai/urbiengine-linux/trunk/src/server-caller.hxx	(revision 453)
+++ /home/thomas/projects/gostai/urbiengine-linux/trunk/src/server-caller.hxx	(working copy)
@@ -13,7 +13,7 @@
 #include <stdexcept>
 #include <sstream>
 // Inclusion order matters for windows. Leave userver.hh after network.hh.
-#include <network/bsdnet/network.hh>
+#include <network/boost/network.hh>
 
 #include <kernel/userver.hh>
 #include <kernel/uconnection.hh>
Index: /home/thomas/projects/gostai/urbiengine-linux/trunk/src/simul-api.hh
===================================================================
--- /home/thomas/projects/gostai/urbiengine-linux/trunk/src/simul-api.hh	(revision 453)
+++ /home/thomas/projects/gostai/urbiengine-linux/trunk/src/simul-api.hh	(working copy)
@@ -36,6 +36,9 @@
       /// Tcp port URBI will listen to.
       int		port;
 
+      /// Use SSL connection.
+      bool		secured;
+
       /// If set, the server try to run as fast as possible and emulate the
       /// interval specified (the "freq" parameter).
       bool		fast_mode;

